rest-service/manager_rest/app_context.py

```diff
@@ -19,23 +43,66 @@ def get_parser_context(sm=None):
 
 
 def update_parser_context(context):
-    raw_parser_context = _extract_parser_context(context)
-    resolver = dsl_parser_utils.create_import_resolver(
-            raw_parser_context['resolver_section'])
-    validate_definitions_version = raw_parser_context[
-        'validate_definitions_version']
-    current_app.parser_context = {
-        'resolver': resolver,
-        'validate_version': validate_definitions_version
-    }
+    current_app.parser_context = _extract_parser_context(context)
 
 
 def _extract_parser_context(context):
     context = context or {}
     cloudify_section = context.get(constants.CLOUDIFY, {})
+    resolver_section = cloudify_section.get(
+        constants.IMPORT_RESOLVER_KEY) or {}
+    resolver_section.setdefault('implementation',
+                                'manager_rest.app_context:ResolverWithPlugins')
+    resolver = dsl_parser_utils.create_import_resolver(resolver_section)
     return {
-        'resolver_section': cloudify_section.get(
-                constants.IMPORT_RESOLVER_KEY),
-        'validate_definitions_version': cloudify_section.get(
-                constants.VALIDATE_DEFINITIONS_VERSION, True)
+        'resolver': resolver,
+        'validate_version': cloudify_section.get(
+            constants.VALIDATE_DEFINITIONS_VERSION, True)
     }
+
+
+class ResolverWithPlugins(DefaultImportResolver):
+    """A resolver which translates plugin-style urls to file:// urls.
+
+    The URL: `file://plugins/cloudify-openstack-plugin/2.0.1` will be
```

Comment:
are u sure about the `file://` prefix?
i was under the impression it was gonna be just:
/plugins/cloudify-openstack-plugin/2.0.1